src/mscorlib/src/System/String.cs

```diff
@@ -1170,6 +1252,35 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
             return foundCount;
         }        
         
+        //--------------------------------------------------------------------
+        // This function returns number of the places within baseString where
+        // instances of the separator string occurs.
+        // Args: separator  -- the separator
+        //       sepList    -- an array of ints for split string indicies.
+        //--------------------------------------------------------------------
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        private unsafe int MakeSeparatorList(string separator, ref int[] sepList) {
+            Contract.Assert(!string.IsNullOrEmpty(separator), "!string.IsNullOrEmpty(separator)");
```

Comment:
Is this better as `Contract.Assert(!string.IsNullOrEmpty(separator), $"!string.IsNullOrEmpty({nameof(separator)})");` or is string interpolation not available?

Also, you seem to bounce between `String` and `string`, which is "correct"?